src/plugins/formulas/formulas.js

```diff
@@ -363,13 +377,13 @@ export class Formulas extends BasePlugin {
    * @param {number} row Visual row index.
    * @param {number} column Visual column index.
    * @param {number} [sheet] The target sheet id, defaults to the current sheet.
-   * @returns {string} Possible values: 'FORMULA' | 'VALUE' | 'MATRIX' | 'EMPTY'.
+   * @returns {string} Possible values: 'FORMULA' | 'VALUE' | 'ARRAYFORMULA' | 'EMPTY'.
    */
   getCellType(row, column, sheet = this.sheetId) {
     return this.engine.getCellType({
       sheet,
-      row: this.hot.toPhysicalRow(row),
-      col: this.hot.toPhysicalColumn(column)
+      row: this.hot.toPhysicalRow(row) ?? Infinity,
+      col: this.hot.toPhysicalColumn(column) ?? Infinity
```

Comment:
IMO this is a bad idea. We're reading `getCellType` from different cell that may be out of bounds

Reply:
The `Infinity` mentions should be removed after https://github.com/handsontable/handsontable/pull/8669/commits/d96ae619e5822d099e6120b57403f4785d3c831f.